skbio/io/_base.py

```diff
@@ -185,3 +186,21 @@ def _format_fasta_like_records(generator, id_whitespace_replacement,
                 "scores associated with it." % cardinal_to_ordinal(idx + 1))
 
         yield header, seq.sequence, seq.quality
+
+
+def _line_generator(fh, skip_blanks=False):
+    for line in fh:
+        line = line.strip()
+        if line or not skip_blanks:
+            yield line
+
+
+def _count_blank_lines(fh):
```

Comment:
This still can take a full pass through the file, which we don't want to happen when sniffing.